shielder/contract/drink_tests/mod.rs

```diff
@@ -0,0 +1,138 @@
+mod utils;
+
+use drink::{session::Session, AccountId32};
+
+use crate::{
+    drink_tests::utils::{chain::*, ops::*, psp22::*, shielder::*},
+    test_utils::merkle::MerkleTree,
+};
+
+#[drink::contract_bundle_provider]
+pub enum BundleProvider {}
+
+#[drink::test]
+fn deploy_single_deposit_single_withdraw(
+    mut session: Session,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let alice: AccountId32 = init_alice(&mut session)?;
+    let bob: AccountId32 = init_bob(&mut session)?;
+
+    session = session.with_actor(alice.clone());
+
+    let shielder_address = deploy_shielder(&mut session)?;
+    let psp22_address = deploy_test_token(&mut session, 100)?;
+
+    let mut merkle_tree = MerkleTree::new();
+
+    // CREATE ACCOUNT
+    let user_shielded_data = create_shielder_account(
+        &mut session,
+        &shielder_address,
+        &psp22_address,
+        &mut merkle_tree,
+    )?;
+
+    // APPROVE TRANSFER
+    psp22_approve(&mut session, &psp22_address, &shielder_address, 10)?;
+
+    // DEPOSIT
+    let user_shielded_data = shielder_update(
+        &mut session,
+        &shielder_address,
+        deposit_op(&psp22_address, &alice, 10),
+        user_shielded_data,
+        &mut merkle_tree,
+    )?;
+
+    let alice_psp22_balance: u128 = get_psp22_balance(&mut session, &psp22_address, &alice)?;
```

Comment:
nit: Maybe it is just my opinion, but I don't think that specifying variable types is helpful and potentially makes refactors harder

Reply:
fixed